public/docs/ts/latest/guide/security.jade

```diff
@@ -0,0 +1,248 @@
+  interpolation, Angular will sanitize and escape untrusted values.
+
+  ### Sanitization and security contexts
+
+  Sanitization inspects an untrusted value and turns it into a value that is safe to insert into
+  the DOM. In many cases, values do not get changed by this at all. Sanitization depends on context:
+  a value that is harmless in CSS is potentially dangerous in a URL.
+
+  Angular defines four security contexts: HTML, style, URL, and resource URL.
+
+  * HTML is used when interpreting a value as HTML, e.g. when binding to `innerHtml`
+  * Style is used when binding CSS into the `style` property
+  * URL is used for URL properties such as `<a href>`
+  * Resource URLs are URLs that will be loaded and executed as code, e.g. in `<script src>`
+
+  Angular sanitizes untrusted values for the first three items; sanitizing resource URLs is not
+  possible as they contain arbitrary code. In development mode, Angular prints a console warning
+  when it has to change a value during sanitization.
+
+  ### Sanitization example
+
+  The template below binds the value of `htmlSnippet`, once by interpolating it into an element's
+  content, and once by binding it to the `innerHTML` property of an element.
+
++makeExample('security/ts/app/inner-html-binding.component.html')(format=".")
+:marked
+  Interpolated content is always escaped - the HTML is not interpreted, and the browser displays
+  angle brackets in the elements text content.
+
+  For the HTML to be interpreted, we must bind to an HTML property, such as `innerHTML`. But binding
+  a potentially attacker controlled value into `innerHTML` would normally cause an XSS
+  vulnerability. For example, code contained in a `<script>` tag would be executed.
++makeExample('security/ts/app/inner-html-binding.component.ts', 'inner-html-controller')(format=".")
+:marked
+  Angular recognizes the value as unsafe, and automatically sanitizes it. It removes the `<script>`
+  tag but keeps safe content, such as the text content of the `<script>` tag, or the `<b>` element.
+
+figure.image-display
+    img(src='/resources/images/devguide/security/binding-inner-html.png'
+        alt='A screenshot showing interpolated and bound HTML values')
+:marked
+  ### Avoid direct use of the DOM APIs
+
+  The built-in browser DOM APIs do not automatically protect you from security vulnerabilities.
+  For example, `document`, the node available through `ElementRef`, and many third party APIs
+  contain unsafe methods. Avoid directly interacting with the DOM, and instead use Angular
+  templates where possible.
+
+  ### Content Security Policy
+
+  A [Content Security Policy (CSP)](https://developer.mozilla.org/en-
+  US/docs/Web/Security/CSP/Introducing_Content_Security_Policy) is a defense-in-depth technique to
+  prevent XSS. To enable CSP, configure your web server to return an appropriate
+  `Content-Security-Policy` HTTP header. Learn more at
+  [OWASP](https://www.owasp.org/index.php/Content_Security_Policy).
```

Comment:
I'm not sure if OWASP is the best source for an introduction to CSP. For example this is better: http://www.html5rocks.com/en/tutorials/security/content-security-policy/ @rjamet @koto do you know any good up to date CSP introduction article?